src/main/java/uk/gov/pay/api/agreement/resource/AgreementsApiResource.java

```diff
@@ -65,10 +111,30 @@ public Response createAgreement(
     @GET
     @Path("/v1/agreements/{agreementId}")
     @Produces(APPLICATION_JSON)
+    @Operation(security = {@SecurityRequirement(name = "BearerAuth")},
+            operationId = "Get an agreement",
+            summary = "Get information about a single agreement",
+            description = "You can use this endpoint to get information about a single recurring payments agreement.",
+            responses = {
+                    @ApiResponse(responseCode = "200", description = RESPONSE_200_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = Agreement.class))),
+                    @ApiResponse(responseCode = "401",
+                            description = RESPONSE_401_DESCRIPTION),
+                    @ApiResponse(responseCode = "404", description = RESPONSE_404_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = RequestError.class))),
+                    @ApiResponse(responseCode = "429", description = RESPONSE_429_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ApiErrorResponse.class))),
+                    @ApiResponse(responseCode = "500", description = RESPONSE_500_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = RequestError.class)))
+            }
+    )
     public Agreement getAgreement(
             @Parameter(hidden = true) @Auth Account account,
-            @PathParam("agreementId") String agreementId
-    ) {
+            @PathParam("agreementId") 
+            @Parameter(name = "agreementId", 
+                    description = "Returns the agreement with the matching `agreement_id`", 
```

Comment:
Should this also include "GOV.UK Pay generated an agreement_id when you created the agreement." from the API reference page?